### PR TITLE
Add provider-native snapshot support 

### LINKED ISSUE #529: Support vagrant 1.8 snapshots
Vagrant 1.8 added a feature to make snapshots.

from https://github.com/mitchellh/vagrant/blob/master/CHANGELOG.md:

```
Snapshots: The vagrant snapshot command can be used to checkpoint and restore point-in-time snapshots.
```

Using latest vagrant-libvirt release (0.0.32), i get: 

```
--- vms/jessie » vagrant snapshot list  
==> jessie: This provider doesn't support snapshots.
==> jessie: 
==> jessie: This may be intentional or this may be a bug. If this provider
==> jessie: should support snapshots, then please report this as a bug to the
==> jessie: maintainer of the provider.

```

I'm using sahara for working with snapshots, so this is not urgent, but a feature request.

### PR DESCRIPTION
**Pull request by randomvariable**
Resolves #529

### CODE REVIEW

File: lib/vagrant-libvirt/action.rb [hunk 1]
```diff
@@ -317,6 +317,58 @@ def self.action_ssh_run
...
+            else
+              b2.use MessageNotCreated
+            end
+
+            b2.use Call, IsRunning do |env2, b3|
+              unless env2[:result]
+                b3.use MessageNotRunning
+                next
+              end
+              b3.use SnapshotDelete
```
Comment (electrofelix) [inline lib/vagrant-libvirt/action.rb]: Is this intended to automatically delete snapshots associated with machines already deleted during the restore?

File: lib/vagrant-libvirt/errors.rb [hunk 2]
```diff
@@ -145,8 +145,24 @@ class NoIpAddressError < VagrantLibvirtError
...
-        error_key(:delete_snapshot_error)
+      class SnapshotMissing < VagrantLibvirtError
+        error_key(:snapshot_missing)
+      end
+
+      class SnapshotDeletionError < VagrantLibvirtError
+        error_key(:snapshot_deletion_error)
+      end
+
+      class SnapshotListnrror < VagrantLibvirtError
```
Comment (electrofelix) [inline lib/vagrant-libvirt/errors.rb]: Typo? `SnapshotListnrror` should be `SnapshotListError`?

**Comment by jpeeler** [comment]
@randomvariable Any chance you could circle back on this? /me just a random guy wanting this feature

**Comment by randomvariable** [comment]
Sorry I dropped the ball on this and moved onto other stuff. I'll try and get it done in next few days.

**Comment by jpeeler** [comment]
@randomvariable Thought I'd try one more bump.

**Comment by gbritton** [comment]
Is there any update on this?  It would be really nice to be able to do multiple named snapshots with vagrant-libvirt rather than the single snapshot model sahara provides.

**Comment by randomvariable** [comment]
I don't think I'm going to be touching this any time soon. I haven't been using Vagrant for a long time. 

Feel free to run with the branch and submit a fixed PR.

**Pull request closed without merging**